### PR TITLE
fix: Add PVC delete permission to che service account

### PR DESCRIPTION
**Pull request by dkwon17**
Signed-off-by: David Kwon <dakwon@redhat.com>

### What does this PR do?
This PR adds delete permissions for the Che service account to allow deleting PVCs.

### Screenshot/screencast of this PR

Che service account is now able to delete PVCs:
image

### What issues does this PR fix or reference?

Fixes https://github.com/eclipse/che/issues/19867

### How to test this PR?

Here is the Che operator image: `quay.io/dkwon17/che-operator:19867`

1. Start minikube and deploy Che with the operator with `chectl`
```
minikube start --addons=ingress --vm=true --memory=8192
chectl server:deploy --installer operator -p minikube --che-operator-image=quay.io/dkwon17/che-operator:19867
```

2. Create a workspace, which creates a PVC named `claim-che-workspace` (here the PVC strategy is `common`) in the `admin-che` namespace:
image

3. Delete the PVC using `oc`
```
oc login --token=<che service account token>
oc project admin-che
oc delete pvc claim-che-workspace
```
The PVC should be deleted.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by tolusha — APPROVED** [review]
Sounds good. let's wait for a green light and merge

**Review by ibuziuk — APPROVED** [review]

**Comment by ibuziuk** [comment]
@tolusha got a question, that is not directly related. Why che-operator does not have permissions to delete PVC - https://github.com/dkwon17/che-operator/blob/5a64f0ad84a27d66c36f76987b04a0aab4fc6bc0/deploy/cluster_role.yaml#L191-L197 ?  is it expected

**Comment by tolusha** [comment]
@ibuziuk 
They come from [1], but I do agree it is better to add them explicitly there. 
@dkwon17 could you cluster_roles.yaml ? Dont forget to run `olm/update-resources.sh` after that and add changed files to the PR

[1] https://github.com/dkwon17/che-operator/blob/5a64f0ad84a27d66c36f76987b04a0aab4fc6bc0/deploy/cluster_role.yaml#L320-L329

**Comment by AndrienkoAleksandr** [comment]
Strange, I thought, that you need also provide permission for che-operator to delete persisted volume claims https://github.com/eclipse-che/che-operator/blob/main/deploy/cluster_role.yaml#L193 .  After that you code delegation provided in this pr should work. Fix me, please, if I'm wrong, but if che-operator itself doesn't have this delete permission, then it can't delegate this permission thougth code to che-server.

**Comment by AndrienkoAleksandr** [comment]
Oh, sorry, @tolusha had already provided the same feedback https://github.com/eclipse-che/che-operator/pull/847#issuecomment-849391713

**Comment by dkwon17** [comment]
Awesome, thanks everyone for the feedback. I pushed the changes to `cluster_role.yaml` and ran `olm/update-resources.sh`.

Just want to confirm, is it normal for `olm/update-resources.sh` to not print any output? I also noticed that git did not detect any new files after running `olm/update-resources.sh` (not sure if there should be OLM files I need to push?)
image

**Comment by tolusha** [comment]
It failed.
Try 
```
set -x
olm/update-resources.sh
```

**Comment by dkwon17** [comment]
Thank you, I ran `olm/update-resources.sh` and pushed the updated files.

File: Dockerfile
Comment (dkwon17) [inline Dockerfile:13]: I'm not sure why running `olm/update_resources.sh` made this change?

Comment (tolusha) [reply Dockerfile:13]: It is fixed in main branch. Pls rebase.

Comment (tolusha) [reply Dockerfile:13]: Probably regexp doesn't work correctly ^(

**Comment by tolusha** [comment]
Fixed. Let's wait for a green light and merge.

**Pull request merged**
